src/v1/client/LPNClientV1.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ILPNClientV1} from "../interfaces/ILPNClientV1.sol";
+import {ILPNRegistryV1} from "../interfaces/ILPNRegistryV1.sol";
+import {QueryOutput} from "../Groth16VerifierExtensions.sol";
+
+error CallbackNotAuthorized();
+
+abstract contract LPNClientV1 is ILPNClientV1 {
+    ILPNRegistryV1 public lpnRegistry;
+
+    modifier onlyLagrangeRegistry() {
+        if (msg.sender != address(lpnRegistry)) {
+            revert CallbackNotAuthorized();
+        }
+        _;
+    }
+
+    constructor(ILPNRegistryV1 _lpnRegistry) {
+        lpnRegistry = _lpnRegistry;
+    }
+
+    function lpnCallback(uint256 requestId, QueryOutput memory result)
+        external
+        onlyLagrangeRegistry
```

Comment:
We might pass via another service one of these days to submit reply on chain, like proofs aggregation networks, and in this case, it wouldn't be our contract but maybe another contract. 
Not important for now I guess but just wanted to share perspective.

Reply:
The user passes in the address of the registry/verifier contract when deploying their contract. So they would just do that and their queries + results would come from that contract